beginend.el

```diff
@@ -171,9 +170,28 @@ BEGIN-BODY and END-BODY are two `progn' expressions passed to respectively
                      (equal dired-details-state 'hidden)))
         ;; 1 line containing directory size
         (setf move (- move 1)))
-      (dired-next-line move)))
+      (dired-next-line move)
+      (while (and (not (eobp))
+                  (member (dired-get-filename 'no-dir t) '("." "..")))
+        (dired-next-line 1))))
   (progn
-    (beginend--goto-nonwhitespace)
+    (let ((stop 3))
+      (when (or (and (boundp 'dired-hide-details-hide-information-lines)
+                     dired-hide-details-hide-information-lines
+                     (boundp 'dired-hide-details-mode)
+                     dired-hide-details-mode)
```

Comment:
We could simplify it even further, thanks for the suggestion. What do you think about:

```emacs
(defun beginend--dired-on-file-p ()
  "Return non-nil if point is on a file.
`.' and `..' are not considered files."
  (condition-case nil
      (dired-get-filename)
    (error nil)))

(beginend-define-mode dired-mode
  (progn
    (while (and (not (eobp))
                (not (beginend--dired-on-file-p)))
      (dired-next-line 1)))
  (progn
    (while (and (not (bobp))
                (not (beginend--dired-on-file-p)))
      (dired-previous-line 1))
    (dired-move-to-filename)))
```

Reply:
If you like it, feel free to update your PR with this code.

Reply:
If we go for something that simple, I would like to see a *much* simpler test suite as well, mostly focused on testing `beginend--dired-on-file-p`. If you agree, please remove the mess :-).